### ISSUE TITLE
Spot Instances /  Preemptible instances

### ISSUE THREAD

**Issue by tylerpotts** [Quansight/qhub#193]
Example GCP config below:
```
google_cloud_platform:
  node_groups:
    spot-high-memory:
      instance: e2-highmem-8
      min_nodes: 0
      max_nodes: 10
      preemptible: true
profiles:
  dask_worker:
    "Spot Large Worker":
      worker_cores_limit: 8
      worker_cores: 7
      worker_memory_limit: 57G
      worker_memory: 57G
      scheduler_memory: 57G
      image: "quansight/qhub-dask-worker:4c8c28332be1fde32f786c54a5961b5f3d789e16"
      scheduler_extra_pod_config:
        nodeSelector:
          "cloud.google.com/gke-nodepool": spot-high-memory
      worker_extra_pod_config:
        nodeSelector:
          "cloud.google.com/gke-nodepool": spot-high-memory
```

Configuration for AWS still requires investigation

**Comment by laisbsc** [comment]
@tylerpotts found a way to use rich's AWS configuration to sort out this issue, although I am not sure what needs to be done.

**Comment by alimanfoo** [comment]
Hiya, can I ask if qhub supports preemptible nodes for dask workers in GCP? Or is this still todo?

**Comment by alimanfoo** [comment]
> Hiya, can I ask if qhub supports preemptible nodes for dask workers in GCP? Or is this still todo?

Ignore me, yes this works, thanks for such an awesome project!

**Comment by rsignell** [comment]
How can I specify spot instances for the Dask workers on a Nebari AWS deployment?

**Comment by rsignell** [comment]
@iameskild I'm pretty sure you told me at some point this was possible -- I was reminded because with Coiled you can do even do "spot_with_fallback" which allows you to request spot and if there are no spot, fails over to on deman, which is extra cool.  
```python
import coiled

cluster = coiled.Cluster(
    region="us-west-2",
    compute_purchase_option="spot_with_fallback",
    arm=True,
    scheduler_port=443,
    n_workers=10,
    account='dask'
)

client = cluster.get_client()
```
 But still *really cool* would be to just be able to specify spot for dask workers.   😄

**Comment by iameskild** [comment]
Hey @rsignell, this is possible on GCP but I think we will need to enable this feature for AWS. (On AWS, they are referred to as "spot instances")

**Comment by rsignell** [comment]
I see above the tantalizing comment: "@tylerpotts found a way to use rich's AWS configuration to sort out this issue, although I am not sure what needs to be done."

**Comment by rsignell** [comment]
@iameskild , I tried going into the AWS console and just editing the eks autoscaling group for the dask workers, adding an additional instance type, and telling it to use 80% spot and 20% on demand.   It seemed to work -- the cluster started and I saw spot requests being fulfilled.  

Any potential problems with this approach? 

![autoscale1](https://github.com/nebari-dev/nebari/assets/125569335/16f4f216-525a-4b9e-9938-341138ce4d22)
![autoscale2](https://github.com/nebari-dev/nebari/assets/125569335/3d8fdf19-bf0a-4378-984d-c10f4e81f855)
